lib/engine/game/g_18_ny/game.rb

```diff
@@ -383,6 +403,72 @@ def upgrade_cost(tile, _hex, entity, spender)
           terrain_cost - discounts
         end
 
+        def route_distance(route)
+          # Count hex edges
+          route.chains.sum { |conn| conn[:paths].each_cons(2).sum { |a, b| a.hex == b.hex ? 0 : 1 } }
+        end
+
+        def route_distance_str(route)
+          "#{route_distance(route)}H"
+        end
+
+        def check_distance(route, _visits)
+          limit = route.train.distance
+          distance = route_distance(route)
+          raise GameError, "#{distance} is too many hex edges for #{route.train.name} train" if distance > limit
+        end
+
+        def revenue_for(route, stops)
+          super + (stops.count { |stop| stop.hex.assigned?('connection_bonus') } * 10)
+        end
+
+        def revenue_str(route)
+          str = super
+
+          if (num_bonuses = route.stops.count { |stop| stop.hex.assigned?('connection_bonus') }).positive?
+            str += " + #{num_bonuses} Connection Bonus#{num_bonuses == 1 ? '' : 'es'}"
+          end
+
+          str
+        end
+
+        def routes_revenue(routes)
+          revenue = super
+          revenue += connection_bonus(routes.first.corporation) if routes.any?
```

Comment:
unless empty

Reply:
Okay. I thought we determined performance was the same though?

Reply:
performance is not the same

Reply:
Hmm...I thought Jen ran it both ways and determined it was. Maybe I'm misremembering.